gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDataset.java

```diff
@@ -135,15 +124,27 @@ protected Iterator<FileSet<CopyEntity>> createFileSets(FileSystem targetFs, Copy
    * table replication.
    */
   @VisibleForTesting
-  Collection<CopyEntity> generateCopyEntities(FileSystem targetFs, CopyConfiguration configuration) throws IOException {
+  Collection<CopyEntity> generateCopyEntities(FileSystem targetFs, CopyConfiguration copyConfig) throws IOException {
     String fileSet = this.getFileSetId();
     List<CopyEntity> copyEntities = Lists.newArrayList();
     Map<Path, FileStatus> pathToFileStatus = getFilePathsToFileStatus();
     log.info("{}.{} - found {} candidate source paths", dbName, inputTableName, pathToFileStatus.size());
 
-    for (CopyableFile.Builder builder : getCopyableFilesFromPaths(pathToFileStatus, configuration, targetFs)) {
-      CopyableFile fileEntity =
-          builder.fileSet(fileSet).datasetOutputPath(targetFs.getUri().getPath()).build();
+    Configuration defaultHadoopConfiguration = new Configuration();
+    for (Map.Entry<Path, FileStatus> entry : pathToFileStatus.entrySet()) {
+      Path srcPath = entry.getKey();
+      FileStatus srcFileStatus = entry.getValue();
+      // TODO: determine whether unnecessarily expensive to repeatedly re-create what should be the same FS: could it
+      // instead be created once and reused thereafter?
+      FileSystem actualSourceFs = getSourceFileSystemFromFileStatus(srcFileStatus, defaultHadoopConfiguration);
```

Comment:
If we have assumption that all files for source are on same fs, we should be safe to do so